webapp/src/test/java/com/box/l10n/mojito/service/branch/notification/BranchNotificationServiceTest.java

```diff
@@ -233,4 +252,108 @@ void translateBranch(Branch branch) throws ExecutionException, InterruptedExcept
     // make sure the stats are ready for whatever is done next in notification
     branchStatisticService.computeAndSaveBranchStatistics(branch);
   }
+
+  @Test
+  public void safeTranslationNotification() throws Exception {
+    BranchTestData branchTestData = new BranchTestData(testIdWatcher);
+
+    String branch2ContentUpdated = "# string1 description\nstring1=content1\n";
+
+    AssetContent assetContentBranch2 =
+        assetContentService.createAssetContent(
+            branchTestData.getAsset(), branch2ContentUpdated, false, branchTestData.getBranch2());
+    assetExtractionService
+        .processAssetAsync(assetContentBranch2.getId(), null, null, null, null)
+        .get();
+
+    // New message must be sent first for any translated message to go out
+    waitForCondition(
+        "Branch2 new notification must be sent",
+        () ->
+            branchNotificationRepository
+                    .findByBranchAndNotifierId(branchTestData.getBranch2(), branchNotifierId)
+                    .getNewMsgSentAt()
+                != null);
+
+    // Branch is tracked for safe i18n if it has a merge target. It shouldn't send the translation
+    // until the commit exists or the translated time is over the default 8 hour mark
+    BranchMergeTarget branchMergeTarget = new BranchMergeTarget();
+    branchMergeTarget.setBranch(branchTestData.getBranch2());
+    branchMergeTarget.setTargetsMain(true);
+    branchMergeTarget = branchMergeTargetRepository.save(branchMergeTarget);
+
+    translateBranch(branchTestData.getBranch2());
+
+    // Wait for branch stats to run
+    Thread.sleep(1000);
```

Comment:
Everytime a long sleep (1sec) like this is introduced, out test suite gets that bit slower :(.

Could you think of any alternative to doing this long sleep? I.e.: maybe triggering a branch stats executions synchronously here or something else?

Reply:
And some for the other thread sleep statements in this test